discoveryservice/store.go

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package discoveryservice
+
+import (
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/discoveryservice/log"
+	credential "github.com/nuts-foundation/nuts-node/vcr/credential"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+	"gorm.io/gorm/schema"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var ErrServiceNotFound = errors.New("discovery service not found")
+var ErrPresentationAlreadyExists = errors.New("presentation already exists")
+
+type serviceRecord struct {
+	ID        string `gorm:"primaryKey"`
+	Timestamp uint64
+}
+
+func (s serviceRecord) TableName() string {
+	return "discoveryservices"
+}
+
+var _ schema.Tabler = (*presentationRecord)(nil)
+
+type presentationRecord struct {
```

Comment:
what does gorm do with respect to snakecasing and case-sensitivity?

Reply:
snake casing (becomes `presentation_records`), implemented `Tabler` to customize table names